GitHub Action to release new homebrew versions

jamiew opened the pull request:
Using https://github.com/marketplace/actions/bump-homebrew-formula

I found a few other similar actions but this one seems solid

I created a COMMITTER_TOKEN that only has `public_repo` permission, but from what I understand, if I give that `push` permission, it can automatically update the Homebrew repo

sapegin reviewed (approved):
Awesome! Let's try with a pull request and see what it does -- we can enable direct releases later ;-)

jamiew committed:
Add workflow to auto-update Homebrew formula on release

Triggers on GitHub release publish, uses mislav/bump-homebrew-formula-action@v3
to open a PR against git-friendly/homebrew-git-friendly with the new version URL and SHA.
